### PR TITLE
Modularization pt 1: Networking

### PR DESCRIPTION
**Pull request by DStrand1**
Reorganizes the networking code.

Also creates `gregtech.core`, which will gradually take more and more out of both `gregtech.api` and `gregtech.common` (though mostly api) to slowly reorganize the codebase

Aspects of this PR that are not final:
- Packet registration will eventually be done in the Core Module, though for now it is simply being done in the main mod class

In the next PR:
- Modules, but without any of the user-facing configuration options (yet). This will let the code be organized how it will end up, but not show incomplete features to players

### CODE REVIEW

**Review by LAGIdiot — CHANGES_REQUESTED** [review]
I have put comments on places regarding packets which does not seems right. Also order of imports on some files is not correct.

File: src/main/java/gregtech/core/network/packets/PacketProspecting.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package gregtech.api.net.packets;
+package gregtech.core.network.packets;
```
[the remark this answers is not in the document]
Comment (DStrand1) [reply src/main/java/gregtech/core/network/packets/PacketProspecting.java:1]: PacketProspecting is handled differently right now, I would like to refactor it but it does not use the typical networking handler at the moment

File: src/main/java/gregtech/core/network/internal/NetworkHandler.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        boolean hasClientExecutor = IClientExecutor.class.isAssignableFrom(packetClass);
+
+        if (hasServerExecutor == hasClientExecutor) {
+            if (hasServerExecutor) {
+                GTLog.logger.error("Could not register packet {}, as it is both a Server and Client executor! Only one allowed. Skipping...", packetClass.getName());
+            } else {
+                GTLog.logger.error("Could not register packet {}, as it does not have an executor! Must have either IServerExecutor OR IClientExecutor. Skipping...", packetClass.getName());
+            }
+            return;
+        }
```
Comment (LAGIdiot) [inline src/main/java/gregtech/core/network/internal/NetworkHandler.java:57]: It works but its quite hard to read and understand on first try.

I would simplify it to:
```suggestion
        if (hasServerExecutor && hasClientExecutor) {
            GTLog.logger.error("Could not register packet {}, as it is both a Server and Client executor! Only one allowed. Skipping...", packetClass.getName());
            return;
        } 
            
        if (!hasServerExecutor && !hasClientExecutor) {
             GTLog.logger.error("Could not register packet {}, as it does not have an executor! Must have either IServerExecutor OR IClientExecutor. Skipping...", packetClass.getName());
            return;
        }
```

File: src/main/java/gregtech/api/block/BlockCustomParticle.java [hunk 3]
```diff
@@ -1,9 +1,9 @@
 package gregtech.api.block;
 
 import codechicken.lib.vec.Vector3;
-import gregtech.api.net.NetworkHandler;
-import gregtech.api.net.NetworkUtils;
-import gregtech.api.net.packets.SPacketBlockParticle;
+import gregtech.api.GregTechAPI;
+import gregtech.core.network.NetworkUtils;
```
Comment (LAGIdiot) [inline src/main/java/gregtech/api/block/BlockCustomParticle.java:5]: Incorrect order of imports. All committed files should be checked. I think you can set IDEA to do this automatically.

**Review by LAGIdiot — APPROVED** [review]
Factually it is correct so I will approve it. But I had still add some suggestions.

**Review by LAGIdiot — APPROVED** [review]
Nice and clean, thanks!

**Pull request merged**